SCSH_3C_iOS/SCSH_3C_iOS/Modules/Explore/View/BoardSectionCollectionViewCell.swift

```diff
@@ -0,0 +1,202 @@
+            collectionView.topAnchor.constraint(equalTo: contentView.topAnchor),
+            collectionView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor),
+            collectionView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor),
+            collectionView.bottomAnchor.constraint(equalTo: contentView.bottomAnchor),
+        ])
+        
+        // pageControl
+        pageControl.translatesAutoresizingMaskIntoConstraints = false
+        contentView.addSubview(pageControl)
+        NSLayoutConstraint.activate([
+            pageControl.centerXAnchor.constraint(equalTo: contentView.centerXAnchor),
+            pageControl.bottomAnchor.constraint(equalTo: contentView.bottomAnchor),
+        ])
+        
+        // previousbutton
+        previousButton.translatesAutoresizingMaskIntoConstraints = false
+        contentView.addSubview(previousButton)
+        NSLayoutConstraint.activate([
+            previousButton.widthAnchor.constraint(equalToConstant: 30),
+            previousButton.heightAnchor.constraint(equalToConstant: 30),
+            previousButton.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 10),
+            previousButton.centerYAnchor.constraint(equalTo: contentView.centerYAnchor),
+        ])
+        
+        // nextbutton
+        nextButton.translatesAutoresizingMaskIntoConstraints = false
+        contentView.addSubview(nextButton)
+        NSLayoutConstraint.activate([
+            nextButton.widthAnchor.constraint(equalToConstant: 30),
+            nextButton.heightAnchor.constraint(equalToConstant: 30),
+            nextButton.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -10),
+            nextButton.centerYAnchor.constraint(equalTo: contentView.centerYAnchor),
+        ])
+    }
+    
+    private func setupCollectionView() {
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        
+        collectionView.register(BoardCollectionViewCell.self, forCellWithReuseIdentifier: "BoardCollectionViewCell")
+        (collectionView.collectionViewLayout as? UICollectionViewFlowLayout)?.scrollDirection = .horizontal
+        collectionView.isPagingEnabled = true
+        collectionView.showsHorizontalScrollIndicator = false
+    }
+    
+    private func setupPageControl() {
+        pageControl.numberOfPages = items.count
+        pageControl.currentPage = 0
+        pageControl.isUserInteractionEnabled = false
+    }
+    
+    private func setupButtons() {
+        previousButton.backgroundColor = .black
+        previousButton.alpha = 0
+        previousButton.setTitle("<", for: .normal)
+        previousButton.addTarget(self, action: #selector(previousButtonDidTap), for: .touchUpInside)
+
+        nextButton.backgroundColor = .black
+        nextButton.alpha = 0
+        nextButton.setTitle(">", for: .normal)
+        nextButton.addTarget(self, action: #selector(nextButtonDidTap), for: .touchUpInside)
```

Comment:
IIUC, the setup logic for previous/next button follow a similar pattern. 

Have you thought about wrapping it in a function or method?

Reply:
I used to not over-encapsulate when I implemented `UI` for the first time,
From my experience, **UI changes frequently all the time !!,** I only do so if I've double-confirmed with the UI/UX designer from the beginning, or if the app version is stable.

In this case, it's `NOT` over-encapsulation, but I still prefer not to do it 😅

Reply:
It makes sense.